org.adempiere.base/src/org/compiere/acct/Doc_MatchInv.java

```diff
@@ -295,7 +296,12 @@ public BigDecimal getBalance()
 			}
 			else
 			{
-				cr.setQty(getQty().multiply(multiplier).negate());
+				if (m_invoiceLine.getC_UOM_ID() != 0)
+				{
+					int precision = MUOM.getPrecision(getCtx(), m_invoiceLine.getC_UOM_ID());
+					cr.setQty(getQty().multiply(multiplier).negate().setScale(precision, RoundingMode.HALF_UP));
+				} else

```

Comment:
hmm .. isn't it not possible to have invoice line with C_UOM_ID == 0 ?

Reply:
Yes, as a match invoice is only created for products, there must be a uom set.
I will remove the 4 checks in this file.